awkward-cpp/include/awkward/io/json.h

```diff
@@ -61,6 +61,12 @@ namespace awkward {
              int64_t initial,
              double resize);
 
+    // Delete copy constructor
+    FromJsonObjectSchema(const FromJsonObjectSchema&) = delete;
+
+    // Delete copy-assignment constructor
+    FromJsonObjectSchema& operator=(FromJsonObjectSchema&) = delete;
```

Comment:
I'm going to betray some ignorance of modern C++.

This `FromJsonObjectSchema` object has a lot of private members, and many of them are `std::vectors`.

https://github.com/scikit-hep/awkward/blob/05ee4a25151a224074e5335db4b2094475b06b47/awkward-cpp/include/awkward/io/json.h#L353-L373

If we set the destructor to `= delete`, the object _does_ delete all of its private members when the `FromJsonObjectSchema` gets deleted, right? I want to be sure that this code change is not introducing a memory leak.

@ianna, do you know if this is the case?

Reply:
Yikes: https://stackoverflow.com/a/40746247/1623645

> Why mark a destructor as delete?
>
> To prevent the destructor from being invoked, of course ;)

I'm pretty sure that this use of `= delete` is causing a memory leak. It might be hard to notice because the depth of the objects created by `FromJsonObjectSchema` is limited by the depth of the JSON schema, though every time a JSON file is read with `schema=...`, another one of these would be created.

Reply:
> Reducing the number of shared libraries to two (one is the Python module and the other is dlopened through ctypes) is ideal.

This PR no-longer does this. It seems like to merge the Python module and shared libraries is not trivial, so I just removed the static libraries instead, and thus changed the linking between `libawkward` and `_ext` to dynamic.

Reply:
> If we set the destructor to = delete, the object does delete all of its private members when the FromJsonObjectSchema gets deleted, right? I want to be sure that this code change is not introducing a memory leak.

I may have made a mistake, but this line 
```
    // Delete copy constructor
    FromJsonObjectSchema(const FromJsonObjectSchema&) = delete;

    // Delete copy-assignment constructor
    FromJsonObjectSchema& operator=(FromJsonObjectSchema&) = delete;
```

Just erases the copy (assignment) _constructors_ so that the default copy (assignment) constructor is not used.

Reply:
Ah! I thought this was deleting the destructors. Deleting the copy constructors is fine. (I don't care if it would copy all of the data upon assignment—I doubt we ever assign a `FromJsonObjectSchema` instance, anyway.)

I must have hallucinated a tilde at the beginning of one of these lines. Since these aren't destructors, it's fine.

This PR can be merged!